### PR TITLE
sql: fix locking issues

### PR DESCRIPTION
**Pull request by mping-exo**
- Don't lock the workflow run table for `insert` stmts
  - We only need to ensure that any rollback exception we see is recoverable
  - The code at the end of the action run should not ever fail (except for process crash), but that shouldn't need any locks (its just insert+delete)
- Ensure debug logs always show the workflow id
- Drop the FK to prevent locking issues

### CODE REVIEW

File: src/ablauf/job/macros.clj [hunk 1]
```diff
@@ -0,0 +1,40 @@
+(ns ablauf.job.macros
+  (:require [clojure.walk :as walk]))
+
+;; copied from https://github.com/aphyr/dom-top/blob/master/src/dom_top/core.clj
+(defrecord Retry [bindings])
+
+(defmacro with-retry
+  "It's really fucking inconvenient not being able to recur from within (catch)
```
Comment (wdullaer) [inline src/ablauf/job/macros.clj:8]:
```suggestion
  "It's really inconvenient not being able to recur from within (catch)
```
I get the sentiment, but this is a public repo, so let's be a bit less colourful :-)

File: src/ablauf/job/sql.clj [hunk 2]
```diff
@@ -215,35 +220,42 @@
...
                   result+ts (process-task context action-fn task)]
-              (jdbc/with-transaction [tx db {:isolation isolation-level}]
-                (log/debugf "Task %s processing done, proceeding: %s" (:task/id task) result+ts)
-                ;; lock the workflow_run row, effectively making it serializable
-                ;; and preventing concurrent txs issues
-                (jdbc/execute-one! tx [(str "select * from workflow_run where id=? for update") wid])
-                (clean-task tx id)
-                (insert-workflow-restart tx wid wuuid [result+ts]))
-
+              ;; unless we crash, we should NEVER
```
Comment (wdullaer) [inline src/ablauf/job/sql.clj:244]: What happens if we do crash?

File: src/ablauf/job/macros.clj (hunk 1 above)
Comment (mping-exo) [reply src/ablauf/job/macros.clj:8]: Hah, I just copied it from the source

File: src/ablauf/job/sql.clj (hunk 2 above)
Comment (mping-exo) [reply src/ablauf/job/sql.clj:244]: Task stays as pending forever, until we call something like `fail-pending-tasks!`

Comment (wdullaer) [reply src/ablauf/job/sql.clj:244]: ok, so this gets handled by the regular restart logic then.
We will basically run the leaf again if it is idempotent.

Comment (mping-exo) [reply src/ablauf/job/sql.clj:244]: yes

**Review by bcachet — APPROVED** [review]

**Pull request merged**